### ISSUE TITLE
Clickhouse: unable to track tables from other database tables.

### ISSUE THREAD

**Issue by soorajshankar** [hasura/graphql-engine#10094]
### Version Information

HGE Docker : hasura/graphql-engine:v2.36.0
Clickhouse Connector: hasura/clickhouse-data-connector:v2.36.1

(Issue is there in v3 clickhouse connector as well)

Server Version: v2.36.0
CLI Version (for CLI related issue):

### Environment

<details>
<summary>Click to expand Docker Compose file</summary> 

  ### Docker compose file 

  ```yaml
version: "3.7"
services:
  redis:
    image: redis:7
    restart: always
  postgres:
    image: postgres:15
    restart: always
    volumes:
      - db_data:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: postgrespassword
  hasura:
    image: hasura/graphql-engine:v2.36.0
    restart: always
    ports:
      - 8080:8080
    environment:
      HASURA_GRAPHQL_ADMIN_SECRET: myadminsecretkey
      HASURA_GRAPHQL_DATABASE_URL: postgres://postgres:postgrespassword@postgres:5432/postgres
      HASURA_GRAPHQL_ENABLE_CONSOLE: "true"
      HASURA_GRAPHQL_DEV_MODE: "true"
      HASURA_GRAPHQL_REDIS_URL: redis://redis:6379
      HASURA_GRAPHQL_RATE_LIMIT_REDIS_URL: "redis://redis:6379"
      HASURA_GRAPHQL_MAX_CACHE_SIZE: "200"
      HASURA_GRAPHQL_METADATA_DEFAULTS: '{"backend_configs":{"dataconnector":{"clickhouse":{"uri":"http://data-connector-agent:8080"}}}}'
    depends_on:
      data-connector-agent:
        condition: service_healthy
  data-connector-agent:
    image: hasura/clickhouse-data-connector:v2.36.1
    restart: always
    ports:
      - 8081:8080
    environment:
      HASURA_GRAPHQL_LOG_LEVEL: debug
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8080/health"]
      interval: 5s
      timeout: 10s
      retries: 5
      start_period: 5s
  clickhouse:
    image: clickhouse/clickhouse-server:23.12.1.1368
    user: "101:101"
    container_name: clickhouse
    hostname: clickhouse
    volumes:
      - ${PWD}/fs/volumes/clickhouse/etc/clickhouse-server/config.d/config.xml:/etc/clickhouse-server/config.d/config.xml
      - ${PWD}/fs/volumes/clickhouse/etc/clickhouse-server/users.d/users.xml:/etc/clickhouse-server/users.d/users.xml
    ports:
      - "127.0.0.1:8123:8123"
      - "127.0.0.1:9000:9000"
volumes:
  db_data:
  ```
</details>
### What is the current behaviour?

Now execute the following 

```sql
# Hasura will be able to track this table as it is on the default db
CREATE TABLE users (
    id UInt64,
    name String,
    age UInt8,
    email String
) ENGINE = MergeTree() ORDER BY id;

# to show case different database not being listed
CREATE DATABASE IF NOT EXISTS test_db;

CREATE TABLE IF NOT EXISTS test_db.employees
(
    `id` Int32,
    `firstName` String,
    `lastName`  String,
)
ENGINE = MergeTree
ORDER BY id
SETTINGS index_granularity = 8192;
```

### What is the expected behaviour?

The table created under test_db should be visible on hasura as untracked tables and it should be able to track this table. 

### How to reproduce the issue?

1. use the above docker compose 
2. run. `./clickhouse client` 
3. past the above SQL file one by one to create database and tables. 
4. Go to Hasura console and connect the clickhouse db with the following config 
5. <img width="1383" alt="image" src="https://github.com/hasura/graphql-engine/assets/8408875/1b3cb6c8-1aad-4381-ac15-bc2fc7c0368b">
6. Now head to untracked tables, non default db table is not visible but users table is visible.  

### Screenshots or Screencast

^

### Keywords

clickhouse non default database tracking

**Comment by BenoitRanque** [comment]
addressed by [clickhouse_gdc_v2#13](https://github.com/hasura/clickhouse_gdc_v2/pull/13)

**Comment by manasag** [comment]
Here is the V3 [commit](https://github.com/hasura/ndc-clickhouse/commit/7c0ee0c7407ece644a4ee183df26b3032d49146e) for the fix.
